core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -486,31 +489,47 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
 
     Authority auth = Authority.fromString(uri.getAuthority());
     if (auth instanceof UnknownAuthority) {
-      // TODO(zac): In Alluxio 2.0 this warning will be upgraded to an exception
-      LOG.warn("Authority \"{}\" is unknown. The client will not be configured with this"
-          + " authority. The authority connection details will be loaded from your client"
-          + " configuration.",
-          auth);
-      mAlluxioHeader = getScheme() + ":///";
+      throw new IOException(String.format("Authority \"%s\" is unknown. The client will not be "
+              + "configured with this authority. The authority connection details will be loaded "
+              + "from the client configuration.", auth.toString()));
     }
 
+    mAlluxioHeader = getScheme() + ":///";
     mUri = URI.create(mAlluxioHeader);
 
+    // Need to initialize the fsContext before connectDetailsMatch and initializeInternal
+    // both methods need to be able to access the Alluxio Configuration from the
+    // fsContext.
+
     Map<String, Object> uriConfProperties = getConfigurationFromUri(uri);
+    AlluxioProperties alluxioProps = ConfigurationUtils.defaults();
+    alluxioProps.merge(uriConfProperties, Source.RUNTIME);
+    mFsContext = FileSystemContext.create(getHadoopSubject(), alluxioProps);
 
     synchronized (INIT_LOCK) {
       if (sInitialized) {
         if (!connectDetailsMatch(uriConfProperties, conf)) {
           LOG.warn(ExceptionMessage.DIFFERENT_CONNECTION_DETAILS.getMessage(
-              FileSystemContext.get().getMasterInquireClient().getConnectDetails()));
+              mFsContext.getMasterInquireClient().getConnectDetails()));
           initializeInternal(uriConfProperties, conf);
         }
       } else {
         initializeInternal(uriConfProperties, conf);
       }
+
+      // Sets the file system.
+      //
       // Must happen inside the lock so that the global filesystem context isn't changed by a
       // concurrent call to initialize.
-      updateFileSystemAndContext();
+      Subject subject = mFsContext.getClientContext().getSubject();
+      if (subject != null) {
```

Comment:
It looks like this is just for logging which subject we are using. Instead of doing it here, we could just do the logging in `getHadoopSubject()`, so here, we can just create a filesystem from the filesystem context.

Reply:
Well if we put it within the `getHadoopSubject` it would be logged every call. Maybe only after filesystem context initialization?

Granted there is only one call to getHadoopSubject but in the case there are more I feel it could just clog up the logs.